libraries/classes/Dbal/DbalInterface.php

```diff
@@ -577,9 +577,21 @@ public function affectedRows($link = DatabaseInterface::CONNECT_USER, bool $getF
      */
     public function getFieldsMeta(ResultInterface $result): array;
 
+    /**
+     * Returns properly quoted string for use in MySQL queries.
+     *
+     * @param string $str  string to be quoted
+     * @param mixed  $link optional database link to use
+     *
+     * @psalm-return non-empty-string
```

Comment:
```suggestion
     * @psalm-return non-empty-string
     * @psalm-taint-escape sql
```
I think it may work 
https://psalm.dev/docs/security_analysis/avoiding_false_positives/